services/integrationtest.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+USAGE="$0 <test-executable> [args...]"
+EXE=${1:?$USAGE}
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+PARENT_PID=$$
+EXIT_STATUS_LOCATION=/dev/shm/integration_test_rc
+echo 1 >${EXIT_STATUS_LOCATION}
+
+function kill_all() {
+    # kill the process tree of the PARENT_PID
+    kill -9 -${PARENT_PID} &> /dev/null
```

Comment:
Manpage on ubuntu or debian does not say anything about the `-` preceding the PID; there is only the special case `-1` for `everything`.  What's the secret?  (-: